OpenRA.Game/MiniYaml.cs

```diff
@@ -241,40 +242,43 @@ static List<MiniYamlNode> FromLines(IEnumerable<string> lines, string filename,
 					}
 
 					if (keyLength > 0)
-						key = line.Substring(keyStart, keyLength).Trim();
+						key = line.Slice(keyStart, keyLength).Trim();
 
 					if (valueStart >= 0)
 					{
-						var trimmed = line.Substring(valueStart, valueLength).Trim();
+						var trimmed = line.Slice(valueStart, valueLength).Trim();
 						if (trimmed.Length > 0)
 							value = trimmed;
 					}
 
 					if (commentStart >= 0 && !discardCommentsAndWhitespace)
-						comment = line.Substring(commentStart);
+						comment = line.Slice(commentStart);
 
 					// Remove leading/trailing whitespace guards
-					if (value != null && value.Length > 1)
+					if (value.Length > 1)
 					{
 						var trimLeading = value[0] == '\\' && (value[1] == ' ' || value[1] == '\t') ? 1 : 0;
 						var trimTrailing = value[value.Length - 1] == '\\' && (value[value.Length - 2] == ' ' || value[value.Length - 2] == '\t') ? 1 : 0;
 						if (trimLeading + trimTrailing > 0)
-							value = value.Substring(trimLeading, value.Length - trimLeading - trimTrailing);
+							value = value.Slice(trimLeading, value.Length - trimLeading - trimTrailing);
 					}
 
 					// Remove escape characters from #
-					if (value != null && value.IndexOf('#') != -1)
-						value = value.Replace("\\#", "#");
+					if (value.Contains("\\#", StringComparison.Ordinal))
+						value = value.ToString().Replace("\\#", "#");
 				}
 
 				if (key != null || !discardCommentsAndWhitespace)
 				{
-					key = key == null ? null : stringPool.GetOrAdd(key, key);
-					value = value == null ? null : stringPool.GetOrAdd(value, value);
-					comment = comment == null ? null : stringPool.GetOrAdd(comment, comment);
+					var keyString = key == null ? null : key.ToString();
+					var valueString = value == null ? null : value.ToString();
```

Comment:
You've removed the `value != null` checks above, suggesting that it shouldn't be needed here either?

Reply:
`ReadOnlySpan<char> key = null;` produces an empty span. i.e. `key.IsEmpty` is true and `key.Length` gives 0.

Therefore we can replace the null checks with not-empty checks - but the existing logic makes those not-empty checks redundant so they can go away. We need a not-empty check later to avoid allocating empty strings though.

Perhaps this would be clearer by assigning `ReadOnlySpan<char> key = default;` and using `var keyString = key.IsEmpty ? null : key.ToString();` instead?

Reply:
Yeah, that would be much clearer 👍

Reply:
Added. Also added some test cases to cover the comment field being empty when expected as that requires `comment == default` to work as desired.